src/commands/createNewProject/ProjectCreateStep/PythonProjectCreateStep.ts

```diff
@@ -4,42 +4,69 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as fse from 'fs-extra';
+import * as path from 'path';
 import * as semver from 'semver';
-import { QuickPickItem } from 'vscode';
-import { IAzureQuickPickOptions, parseError, UserCancelledError } from 'vscode-azureextensionui';
+import { Progress, QuickPickItem } from 'vscode';
+import { IAzureQuickPickOptions, parseError } from 'vscode-azureextensionui';
+import { extensionPrefix } from '../../../constants';
 import { ext } from '../../../extensionVariables';
-import { validateFuncCoreToolsInstalled } from '../../../funcCoreTools/validateFuncCoreToolsInstalled';
+import { getLocalFuncCoreToolsVersion } from '../../../funcCoreTools/getLocalFuncCoreToolsVersion';
 import { localize } from "../../../localize";
 import { cpUtils } from "../../../utils/cpUtils";
+import { confirmOverwriteFile } from '../../../utils/fs';
 import { venvUtils } from '../../../utils/venvUtils';
-import { getGlobalSetting } from '../../../vsCodeConfig/settings';
+import { getGlobalSetting, getWorkspaceSetting } from '../../../vsCodeConfig/settings';
 import { IProjectWizardContext } from '../IProjectWizardContext';
-import { ProjectCreateStepBase } from './ProjectCreateStepBase';
+import { ScriptProjectCreateStep } from './ScriptProjectCreateStep';
 
 const minPythonVersion: string = '3.6.0';
 const maxPythonVersion: string = '3.7.0';
 const minPythonVersionLabel: string = '3.6.x'; // Use invalid semver as the label to make it more clear that any patch version is allowed
 
-export class PythonProjectCreateStep extends ProjectCreateStepBase {
-    private constructor() {
-        super();
-    }
+// Starting after this version, the func cli does not require a virtual environment and comes pre-packaged with the below dependencies
+const oldFuncVersion: string = '2.4.419';
+const oldRequirements: string = `azure-functions==1.0.0b3
+azure-functions-worker==1.0.0b4
+grpcio==1.14.2
+grpcio-tools==1.14.2
+protobuf==3.7.1
+six==1.12.0
+`;
+
+export class PythonProjectCreateStep extends ScriptProjectCreateStep {
+    // tslint:disable-next-line: no-use-before-declare
```

Comment:
Why is this needed?? I must not be understanding this correctly, but I don't see it getting used before you've declared it.

Reply:
tslint wants `pythonGitignore` defined above `PythonProjectCreateStep` in this file. I think that's annoying because I care much more about `PythonProjectCreateStep`